File: frontend/src/types/trade.ts
/**
 * 거래 관련 타입 정의
 */

export interface Trade {
  id: string
  symbol: string
  exchange: string
  side: 'buy' | 'sell'
  type: 'market' | 'limit'
  amount: number
  price: number
  cost: number
  fee: number
  status: 'open' | 'closed' | 'cancelled' | 'filled' | 'failed'
  is_paper: boolean
  agent_id?: string | null
  strategy?: string
  stop_loss?: number
  take_profit?: number
  /** 매도 체결 시 청산가 */
  close_price?: number | null
  pnl?: number
  created_at: string
  updated_at: string
}

export interface TradeListResponse {
  trades: Trade[]
  total: number
}

File: backend/schemas/trade.py
"""거래 관련 Pydantic 스키마"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class TradeResponse(BaseModel):
    """거래 응답"""
    id: str
    symbol: str
    exchange: str
    side: str
    type: str
    amount: float
    price: float
    cost: float
    fee: float
    status: str
    is_paper: bool
    strategy: Optional[str] = None
    stop_loss: Optional[float] = None
    take_profit: Optional[float] = None
    pnl: Optional[float] = None
    created_at: datetime


class TradeListResponse(BaseModel):
    """거래 목록 응답"""
    trades: list[TradeResponse]
    total: int
